src/OpenConext/Profile/Value/Arp.php

```diff
@@ -186,4 +187,40 @@ public function getMotivationFor(
         }
         return '';
     }
+
+    /**
+     * Accordingly to https://www.pivotaltracker.com/n/projects/1453004/stories/187607790
+     * the ARP data should only contain the following keys: value, source, motivation.
+     * This function will filter out all other keys, which are to be ignored.
+     */
+    private static function sanitizeARPData(array $arp): array
+    {
+        $validKeys = ['value', 'source', 'motivation'];
+
+        $sanitizedArp = [];
+        foreach ($arp as $attributeName => $attributeInformation) {
+            if (!is_array($attributeInformation)) {
+                throw new InvalidArpDataException('The attribute information in the arp should be an array.');
+            }
+
+            $sanitizedAttributeInformation = [];
+            foreach ($attributeInformation as $attributeInformationEntry) {
+                if (!is_array($attributeInformationEntry)) {
+                    continue;
```

Comment:
Is this actually occurring? This feels like a corner case where something bigger is wrong.  Also, given you added this bit of behavior, its only fair to also cover it in test.

Reply:
It is, as it was in the create before.... In the test (see ArpTest line 102) it has thios edge...